enterprise/signals/gp_bases.py

```diff
@@ -232,15 +273,82 @@ def createfourierdesignmatrix_chromatic(toas, freqs, nmodes=30, Tspan=None, logf
     :param fmin: lower sampling frequency
     :param fmax: upper sampling frequency
     :param idx: Index of chromatic effects
+    :param modes: option to provide explicit list or array of
+                  sampling frequencies
 
     :return: F: Chromatic-variation fourier design matrix
     :return: f: Sampling frequencies
     """
 
     # get base fourier design matrix and frequencies
-    F, Ffreqs = createfourierdesignmatrix_red(toas, nmodes=nmodes, Tspan=Tspan, logf=logf, fmin=fmin, fmax=fmax)
+    F, Ffreqs = createfourierdesignmatrix_red(
+        toas, nmodes=nmodes, Tspan=Tspan, logf=logf, fmin=fmin, fmax=fmax, modes=modes
+    )
 
     # compute the DM-variation vectors
     Dm = (1400 / freqs) ** idx
 
     return F * Dm[:, None], Ffreqs
+
+
+@function
+def createfourierdesignmatrix_general(
+    toas,
```

Comment:
Question: is this not just a superset of the regular `createfourierdesignmatrix`? In other words, could it replace it?

Reply:
Yes, it is a superset of "createfourierdesignmatrix_red", "..._dm", "..._dm_tn" and "..._chromatic". Where the "chromatic" is a superset of "dm" by the way.

The "general" also allows to mask toas from chosen flags. It is pretty convenient as it is the only current way to make a selection on spatially correlated common signals (as far as I know).

Reply:
Actually, selecting on common signals is in another PR as well. Have a look:

https://github.com/nanograv/enterprise/pull/290

Reply:
Missed that, very nice ! It still need to be merged if I've understood well.

Reply:
Correct. I wanted to review it the other day, but I wasn't sure whether it was relevant since it had been 2 years. I should ping @vallis again regarding it.